src/android/FileUtils.java

```diff
@@ -73,8 +73,13 @@ Licensed to the Apache Software Foundation (ASF) under one
      * Permission callback codes
      */
 
-    public static final int READ_PERM = 0;
+    public static final int READ_PERM_FILE = 0;
     public static final int WRITE_PERM = 1;
+    public static final int READ_PERM_DIR = 2;
+    public static final int WRITE_PERM_DIR = 3;
+    public static final int WRITE = 4;
+    public static final int READ = 5;
+    public static final int WRITE_PERM_FILE = 6;
```

Comment:
The difference between the `WRITE` flag and the `WRITE_PERM`* flags are a bit confusing. Maybe rename `WRITE` to `WRITE_ACTION` and `READ` to `READ_ACTION`?